homeassistant/components/rki_covid/sensor.py

```diff
@@ -0,0 +1,207 @@
+from homeassistant.helpers import update_coordinator
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import (
+    ConfigType,
+    DiscoveryInfoType,
+    HomeAssistantType,
+)
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import get_coordinator
+from .const import (
+    ATTR_COUNTY,
+    ATTRIBUTION,
+    CONF_BASEURL,
+    CONF_DISTRICT_NAME,
+    CONF_DISTRICTS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+# wait for x minutes after restart, before refreshing
+SCAN_INTERVAL = timedelta(minutes=10)
+
+# schema for each config entry
+DISTRICT_SCHEMA = vol.Schema({vol.Required(CONF_NAME): cv.string})
+
+# schema for each platform sensor
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_BASEURL): str,
+        vol.Required(CONF_DISTRICTS): vol.All(cv.ensure_list, [DISTRICT_SCHEMA]),
+    }
+)
+
+SENSORS = {
+    "count": "mdi:virus",
+    "deaths": "mdi:cross",
+    "recovered": "mdi:bottle-tonic-plus-outline",
+    "weekIncidence": "mdi:clipboard-pulse",
+    "casesPer100k": "mdi:home-group",
+    "newCases": "mdi:shield-bug",
+    "newDeaths": "mdi:shield-cross",
+    "newRecovered": "mdi:shield-sync",
+    "hospitalizationCasesBaby": "mdi:baby-face-outline",
+    "hospitalizationIncidenceBaby": "mdi:baby-face",
+    "hospitalizationCasesChildren": "mdi:account-child-outline",
+    "hospitalizationIncidenceChildren": "mdi:account-child",
+    "hospitalizationCasesTeen": "mdi:face-woman",
+    "hospitalizationIncidenceTeen": "mdi:face-woman-outline",
+    "hospitalizationCasesGrown": "mdi:face-man",
+    "hospitalizationIncidenceGrown": "mdi:face-man-outline",
+    "hospitalizationCasesSenior": "mdi:account-cowboy-hat-outline",
+    "hospitalizationIncidenceSenior": "mdi:account-cowboy-hat",
+    "hospitalizationCasesOld": "mdi:human-white-cane",
+    "hospitalizationIncidenceOld": "mdi:human-cane",
+}
+
+
+async def async_setup_platform(
+    hass: HomeAssistantType,
+    config: ConfigType,
+    async_add_entities: Callable,
+    discovery_info: Optional[DiscoveryInfoType] = None,
+) -> None:
+    """Set up the sensor platform."""
+    _LOGGER.debug("setup sensor for platform")
+    session = async_get_clientsession(hass)
+
+    if CONF_BASEURL in config:
+        _LOGGER.warning("Baseurl is not supported anymore.")
+
+    parser = RkiCovidParser(session)
+    coordinator = await get_coordinator(hass, parser)
+
+    if coordinator is None or coordinator.data is None:
+        raise PlatformNotReady("Data coordinator could not be initialized!")
+
+    districts = config[CONF_DISTRICTS]
+
+    sensors = [
+        RKICovidNumbersSensor(coordinator, district[CONF_DISTRICT_NAME], info_type)
+        for info_type in SENSORS
+        for district in districts
+    ]
+    async_add_entities(sensors, update_before_add=True)
```

Comment:
Remove as we should load the integration from config entry

Reply:
I recall the integration guide recommends adding a configuration setup as well. But I might be wrong here.

Reply:
Don't think it's specifically mentioned in the docs but it's not asked to create yaml setup together with a config flow for new integrations